### PR TITLE
0.2/708 display inactive users

### PR DESCRIPTION
**Pull request by katjam**
Still needs tests - and manual testing - but that will be easier once form is ready.

### CODE REVIEW

File: app/Http/Controllers/Service/FamilyController.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        // Since a family might have more than one Registration we send it with the Request.
+        $reg_family = $registration->family;
+        $family = Family::findOrFail($request->get('id'));
+
+        // Make sure we have a family, a registartion that match.
+        if ($family !== $reg_family) {
+            Log::info($user->id . ' attempted to update a family without a matching registration.');
+            return redirect()
+                ->route('service.registration.edit', $registration->id)
+                ->with('error', 'Family deactivation failed. Please try again.')
```
Comment (charlesstrange2) [inline app/Http/Controllers/Service/FamilyController.php:34]: nice!

File: app/Http/Controllers/Service/FamilyController.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            ;
+        }
+
+        if (Gate::forUser($user)->denies('update_registration', $registration)) {
```
Comment (charlesstrange2) [inline app/Http/Controllers/Service/FamilyController.php:38]: Is this the same as 
Auth::user()->can()?

File: app/Http/Controllers/Service/FamilyController.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            Log::info('Family update ' . $registration->id . ' deactivate family ' . $family->id . ' by user ' . $user->id . ' failed.');
+            return response('Unauthorized.', 401);
+        }
+
+        // Validate the request.
+        $this->validate($request, Family::rules());
```
Comment (charlesstrange2) [inline app/Http/Controllers/Service/FamilyController.php]: You're expecting a datetime from the form. 0/1 better?

**Review by charlesstrange2 — COMMENTED** [review]
You validate for a datetime leaving_on input from the form. Better to have a bool as the form is as scriptless as I can manage?

File: app/Http/Controllers/Service/FamilyController.php (hunk 2 above)
Comment (katjam) [reply app/Http/Controllers/Service/FamilyController.php:38]: Maybe - I thought that would only work for policies - but we should try.

File: app/Http/Controllers/Service/FamilyController.php (hunk 3 above)
Comment (katjam) [reply app/Http/Controllers/Service/FamilyController.php]: Oh yeah. Darn. I guess I have to transform to a date first.

Comment (charlesstrange2) [reply app/Http/Controllers/Service/FamilyController.php]: I put request checking on a formRequest class.

File: app/Http/Controllers/Service/FamilyController.php [hunk 4]
```diff
@@ -41,15 +41,27 @@ public function familyUpdate(Request $request, Registration $registration)
         }
 
         // Validate the request.
+        // Maybe this is messy but... converting $request true to date here before validation.
+        $request->merge([
+            'leaving_on' => $request->leaving_on ? Carbon::now() : null,
```
Comment (charlesstrange2) [inline app/Http/Controllers/Service/FamilyController.php]: It' as messy as my stuff, i guess.

**Review by charlesstrange2 — APPROVED** [review]
Give it a whirl! :shipit:

**Comment by katjam** [comment]
I am under no illusions. It will not be right - but we can test once wired up and amend. :)

**Pull request merged**